lib/notificationHandler.js

```diff
@@ -0,0 +1,297 @@
+/**
+ * Notification handler class
+ *
+ *  2021 foxriver76 <moritz.heusinger@gmail.com>
+ */
+
+const fs = require('fs-extra');
+const tools = require('./tools');
+const path = require('path');
+
+class NotificationHandler {
+
+    /**
+     * Create a new instance of NotificationHandler
+     *
+     * @param {object} settings - settings like states/objects db
+     */
+    constructor(settings) {
+        this.states = settings.states;
+        this.objects = settings.objects;
+        this.log = settings.log;
+        this.currentNotifications = {};
+        this.setup = {};
+        // default data dir is relative to main.js
+        this.dataDir = path.join(__dirname, '..', tools.getDefaultDataDir());
+        this.logPrefix = settings.logPrefix;
+        this.host = settings.host;
+
+        // load notifications file
+        this._loadNotifications();
+    }
+
+    /**
+     * Get all adapter instances on this host and store their notifications config
+     *
+     * @return {Promise<void>}
+     */
+    async getSetupOfAllAdaptersFromHost() {
+        try {
+            const res = await this.objects.getObjectViewAsync('system', 'instance', {
+                startkey: 'system.adapter.',
+                endkey: 'system.adapter.\u9999'
+            });
+
+            if (res && Array.isArray(res.rows)) {
+                for (const entry of res.rows) {
+                    // check that instance is running on given host and has notifications
+                    if (entry && entry.value && entry.value.notifications && entry.value.common && entry.value.common.host === this.host) {
```

Comment:
I would read all - not only from same host because so it is more flexible and "global" when you thing further on adapter usage